pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -350,7 +350,11 @@ func (og *operationGenerator) GenerateAttachVolumeFunc(
 			}
 
 			// On failure, return error. Caller will log and retry.
-			return volumeToAttach.GenerateError("AttachVolume.Attach failed", attachErr)
+			eventErr, detailedErr := volumeToAttach.GenerateError("AttachVolume.Attach failed", attachErr)
+			for _, pod := range volumeToAttach.ScheduledPods {
+				og.recorder.Eventf(pod, v1.EventTypeWarning, "ProvisioningAttachFailed", eventErr.Error())
```

Comment:
The event should be a constant instead of a string literal here. See [event.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/events/event.go#L57)

Reply:
Yes, EventRecorderFunc will record this attachErr and return FailedAttachVolume, in old version , there is no such EventRecorderFunc, i will close this PR.